feature_engine/encoding/count_frequency.py

```diff
@@ -131,7 +131,12 @@ def __init__(
 
         self.encoding_method = encoding_method
 
-    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+    def fit(
+        self,
+        X: pd.DataFrame,
+        y: Optional[pd.Series] = None,
+        threshold: Optional[List[Optional[float]]] = None,
```

Comment:
we can't add parameters to the fit method. By convention, fit takes only X and y.

threshold should be in the init.